catalog/schematron/catalog-schematron-plugin/src/test/java/ddf/services/schematron/SchematronValidationServiceTest.java

```diff
@@ -166,6 +167,83 @@ public void testWithSpacesInPathToSchematronFile()
                 "dog_3leg_3paw.xml"));
     }
 
+    @Test
+    public void testWarningsAndErrorsAreSanitized()
+            throws ValidationException, IOException, SchematronInitializationException {
+        SchematronValidationService service = getService("dog_legs.sch", "dog_paws.sch");
+        try {
+            service.validate(getMetacard("dog_4leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            if (ex.getWarnings() != null) {
```

Comment:
Since this should have warnings and no errors, I think it makes the most sense to remove the part about errors, and not have an `if` for the warnings, just say `for (String warning : ex.getWarnings()) { assertThat(warning.equals(service.sanitize(warning)), is(true));` }